src/Zigzag_persistence/include/gudhi/Zigzag_persistence.h

```diff
@@ -0,0 +1,751 @@
+    matrix_.remove_maximal_face(simplexIndex, {});
+  }
+
+  /**
+   * @brief Returns the current persistence diagram ordered by length without infinit bars.
+   *
+   * @param shortest_interval Intervals shorter than the given value are ignored.
+   * @return Vector of intervals.
+   */
+  std::vector<filtration_value_interval> _get_persistence_diagram(Filtration_value shortest_interval) {
+    std::vector<filtration_value_interval> diag;
+    diag.reserve(persistence_diagram_.size());
+
+    std::stable_sort(filtration_values_.begin(), filtration_values_.end(),
+                     [](std::pair<Simplex_key, Filtration_value> p1, std::pair<Simplex_key, Filtration_value> p2) {
+                       return p1.first < p2.first;
+                     });
+
+    for (auto bar : persistence_diagram_) {
+      Filtration_value birth, death;
+      std::tie(birth, death) = map_index_to_filtration_value(bar.birth(), bar.death());
+      if (birth > death) {
+        std::swap(birth, death);
+      }
+
+      if (death - birth > shortest_interval) {
+        diag.emplace_back(bar.dim(), birth, death);
+      }
+    }
+
+    return diag;
+  }
+
+  /**
+   * @brief Computes the births of the current essential cycles.
+   *
+   * @param diag Reference to vector where to store the infinit bars.
+   */
+  void _retrieve_infinit_bars(std::vector<filtration_value_interval>& diag) const {
+    auto birth = [this](Simplex_key b_key) {
+      auto it_b =  // lower_bound(x) returns leftmost y s.t. x <= y
+          std::lower_bound(
+              filtration_values_.begin(), filtration_values_.end(),
+              std::pair<Simplex_key, Filtration_value>(b_key, std::numeric_limits<Filtration_value>::infinity()),
+              [](std::pair<Simplex_key, Filtration_value> p1, std::pair<Simplex_key, Filtration_value> p2) {
+                return p1.first < p2.first;
+              });
+      if (it_b == filtration_values_.end() || it_b->first > b_key) {
+        --it_b;
+      }
+      return it_b->second;
+    };
+
+    for (auto& p : births_) {
+      auto dim = matrix_.get_column(matrix_.get_column_with_pivot(p.first)).get_dimension();
+      if (dim_max_ == -1 || (dim_max_ != -1 && dim < dim_max_)) 
+        diag.emplace_back(dim, birth(p.second), std::numeric_limits<Filtration_value>::infinity());
+    }
+  }
+
+ private:
+  Complex cpx_;                                     /**< Complex in which the current simplices are stored. */
+  int dim_max_;                                     /**< Maximal dimension of a bar to record. */
+  matrix_type matrix_;                              /**< Matrix storing a base of the current chain complex. */
+  std::unordered_map<index, int> births_;           /**< Map simplex index in F to corresponding birth. */
+  birth_ordering birth_ordering_;                   /**< Maintains <b ordering of the births. */
+  std::list<index_interval> persistence_diagram_;   /**< Stores current closed persistence intervals. */
```

Comment:
Do you specifically need `std::list`, or would `std::vector` work?
(https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#slcon2-prefer-using-stl-vector-by-default-unless-you-have-a-reason-to-use-a-different-container)

Reply:
We needed it at some point (I think we did some `emplace_front` before), but indeed, I don't see any reasons anymore to keep it as a list. 
But I am pretty sure I told myself the same thing a while back already and ended up rollbacking to std::list... To keep in mind.